api/src/build.rs

```diff
@@ -188,3 +194,23 @@ fn extract_tarball(crate_path: &Path, project_path: &Path) -> Result<()> {
         Ok(())
     }
 }
+
+fn check_shuttle_version(working_directory: &Path) -> anyhow::Result<()> {
+    let cargo_path = working_directory.join("Cargo.toml");
+    let cargo_doc = read_to_string(cargo_path)?.parse::<Document>()?;
+    let current_shuttle_version = &cargo_doc["dependencies"]["shuttle-service"]["version"];
+    let service_semver = Version::parse(current_shuttle_version.as_str().unwrap())?;
+    let server_version = Version::parse(shuttle_service::VERSION)?;
+
+    let version_required = format!("{}.{}", server_version.major, server_version.minor);
+    let server_semver = VersionReq::parse(&version_required)?;
```

Comment:
Can you maybe parameterise this? That way we can set, from the command-line arguments, the required version for deployment. If the argument is not given, then use a default.